src/app/core/basic-datatypes/entity/entity.datatype.spec.ts

```diff
@@ -23,27 +23,66 @@ import { EntitySchemaField } from "../../entity/schema/entity-schema-field";
 import { TestEntity } from "../../../utils/test-utils/TestEntity";
 
 describe("Schema data type: entity", () => {
-  testDatatype(new EntityDatatype(null, null), "1", "1", "User");
+  testDatatype(new EntityDatatype(null as any, null as any), "1", "1", "User");
 
-  it("should map to the referenced entity", async () => {
-    const c1 = TestEntity.create("first");
-    const c2 = new TestEntity();
-    c2.other = "123";
-    const entityMapper = mockEntityMapper([c1, c2]);
-    const dataType = new EntityDatatype(entityMapper, null);
-    const schema = TestEntity.schema.get("ref");
+  let entityMapper: ReturnType<typeof mockEntityMapper>;
+  let dataType: EntityDatatype;
+  let schema: EntitySchemaField;
 
+  let c1: TestEntity;
+  let c2: TestEntity;
+
+  beforeEach(() => {
+    c1 = TestEntity.create("first");
+    c2 = new TestEntity();
+    c2.other = "123"; // Ensure other is a string
+    entityMapper = mockEntityMapper([c1, c2]);
+    dataType = new EntityDatatype(entityMapper, null as any);
+    schema = TestEntity.schema.get("ref") as EntitySchemaField;
+  });
+
+  it("should map to the referenced entity by name", async () => {
     await expectAsync(
       dataType.importMapFunction("first", schema, "name"),
     ).toBeResolvedTo(c1.getId());
+  });
+
+  it("should map to the referenced entity by other field", async () => {
     await expectAsync(
       dataType.importMapFunction("123", schema, "other"),
     ).toBeResolvedTo(c2.getId());
+  });
+
+  it("should return undefined when no matching entity is found", async () => {
     await expectAsync(
       dataType.importMapFunction("345", schema, "other"),
     ).toBeResolvedTo(undefined);
   });
 
+  it("should handle numeric-string mismatches correctly", async () => {
```

Comment:
what is the "numeric-string mismatch" here in the test code? I don't see any mismatch - it looks like the same logic as `it("should map to the referenced entity by other field` above?

If so, let's simplify test code and remove one of the (almost) identical test cases. I personally like these last two ("should handle numeric-string mismatches correctly" and "should handle numeric value correctly") because they have exactly the same structure and the only difference in the test code is that aspect that we really want to test, i.e. importing a string or number value.

So let's remove some of the other tests if they don't check for fundamentally different things